commandlineparser/src/Parser.cpp

```diff
@@ -51,21 +52,21 @@ bool cmdParser::Parser::Parse(int argc,char* argv[])
 	{
 		default_help(keys);
 	}
-	
-	auto detect_minus_h=std::find(tokenized_data.begin(), tokenized_data.end(), "-h");
+
+	// display short and long help.
 	auto detect_minus_minus_help = std::find(tokenized_data.begin(), tokenized_data.end(), "--help");
 
 	if (detect_minus_minus_help != tokenized_data.end())
 	{
 		long_help(keys);
-		//To terminate the program.
-		std::exit(0);
 	}
-	else if (detect_minus_h != tokenized_data.end())
+	else
 	{
-		short_help(keys);
-		//To terminate the program.
-		std::exit(0);
+		auto detect_minus_h = std::find(tokenized_data.begin(), tokenized_data.end(), "-h");
+		if (detect_minus_h != tokenized_data.end())
+		{
+			short_help(keys);
+		}
 	}
 	return true;
 }
```

Comment:
std::tuple<std::string,std::function<void(std::vector<string>)> > help_functions {{--help,&long_help},{"-h",&short_help}
for_each( auto help_function : help_function)
{
 if (detect_minus_minus_help != tokenized_data.end())
	{		{
		long_help(keys);			long_help(keys);
		//To terminate the program.	
		std::exit(0);	
	}		}
}